File: 1922. Count Good Numbers/countGoodNumbers.cpp
class Solution {
public:
    long long getPower(long base, long long power, long long mod){
        if(power == 0) return 1;

        long long half = getPower(base, power/2 , mod);
        long long res = (half * half) % mod;
        if(power % 2 == 1) res = (res * base) % mod;

        return res;
    };
    int countGoodNumbers(long long n) {
        long long mod = 1000000007;
        long long even = (n + 1) / 2;
        long long odd = n / 2;

        long long evenPart = getPower(5, even, mod);
        long long oddPart = getPower(4, odd, mod);

        return (evenPart * oddPart) % mod;
    }
};

File: 1922. Count Good Numbers/countGoodNumbers.js
var countGoodNumbers = function (n) {
    const MOD = 1000000007n
    let big_n = BigInt(n)
    const EVEN = (big_n + 1n) / 2n
    const ODD = big_n / 2n

    let evenPart = calcPower(5n, EVEN, MOD)
    let oddPart = BigInt(calcPower(4n, ODD, MOD))

    return Number((evenPart * oddPart) % MOD)
};

function calcPower(base, power, MOD) {
    //Recursion  
    if (power === 0n) return 1;

    let half = calcPower(base, power / 2n, MOD)
    let res = BigInt(half * half) % MOD
    if (power % 2n === 1n) res *= base
    return res;

    //Looping

    // while(power > 0n){
    //     if(power % 2n === 1n){//ODD
    //         res = (res * base) % MOD
    //     }
    //     power /= 2n
    //     base = (base*base) % MOD
    // }

    // return res
}